### PR TITLE
GCS_MAVLink: report sequence based rx loss with SYS_STATUS

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1360,6 +1361,15 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
             hal.util->persistent_data.last_mavlink_msgid = msg.msgid;
             hal.util->perf_begin(_perf_packet);
             packetReceived(status, msg);
+            if (msg.sysid == sysid_my_gcs() && total_msg_cnt != 0) {
+                lost_msg_cnt += uint8_t(msg.seq - last_gcs_msg_seq - 1);
```
Comment (amilcarlucas) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: I do not get why these two line are so identical

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -1360,6 +1361,15 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
...
+                total_msg_cnt += uint8_t(msg.seq - last_gcs_msg_seq);
+                last_gcs_msg_seq = msg.seq;
+            } else if (total_msg_cnt == 0) {
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Maybe make this the first clause so you don't need to repeat the comparison against total_msg_count?

Comment (magicrub) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: agreed, would actually speed it up a tiny bit too

**Comment by tridge** [comment]
seq numbers are per component ID. On the call we had the thought this should be done in check_and_forward() in the mavlink routing code

**Comment by rmackay9** [comment]
We discussed this a bit on the dev call and the outcome was that some of the outstanding review comments haven't been addressed so it's best if those are addressed first.

**Comment by peterbarker** [comment]
@bugobliterator did you still want to proceed with this one?

Keeping track of sequence numbers from each component of `SYSID_MYGCS` is going to be messy.

**Comment by bugobliterator** [comment]
I am not working on it currently. I may come back to it at some point.

**Comment by bugobliterator** [comment]
I think I have lost interest in this.

**Pull request closed without merging**